ceasiompy/ActuatorDisk/func/optimalprop.py

```diff
@@ -0,0 +1,371 @@
+    for r, ctrs, cprs in zip(r, CTrs, CPrs):
+        file.write(f"{r:.7f}     {ctrs:.7f}      {cprs:.7f}     0.0\n")
+
+    file.write("#\n")
+    file.write("MARKER_ACTDISK= Propeller_mirrored_AD_Inlet Propeller_mirrored_AD_Outlet\n")
+    file.write("CENTER= 0.0 -2.0 0.0\n")
+    file.write("AXIS= 1.0 0.0 0.0\n")
+    file.write(f"RADIUS= {radius}    \n")
+    file.write(f"ADV_RATIO= {advanced_ratio}   \n")
+    file.write(f"NROW= {stations}   \n")
+
+    file.close()
+
+
+def thrust_calculator(
+    stations,
+    total_thrust_coefficient,
+    radius,
+    hub_radius,
+    advanced_ratio,
+    free_stream_velocity,
+    prandtl,
+    blades_number,
+):
+
+    """Performing of a calculation to obtain thrust and power coefficients distribution
+
+    Args:
+            stations (float): Number of elements for blade discretization [-]
+            total_thrust_coefficient (float): Total thrust coefficient[-]
+            radius (float): Blade radius [m]
+            hub_radius (float): Radius of the blade at the hub [m]
+            advanced_ratio (float): Free_stream_velocity/(rotational_velocity*diameter)[-]
+            free_stream_velocity (float): Cruise velocity [m/s]
+            prandtl (bool): Correction for tip losses
+            blades_number (int): Blades propeller number[-]
+
+    Returns:
+        dCt_optimal (float): thrust coefficient at every radius [-]
+        dCp (float): power coefficient at every radius [-]
+        r (float): adimensional radius [-]
+    """
+
+    log.info("-------------- Check input values choseen --------------")
+    log.info(f"Number of station= {stations}")
+    log.info(f"Selected total thrust coeff= {total_thrust_coefficient}")
+    log.info(f"Radius= {radius}")
+    log.info(f"Hub radius= {hub_radius}")
+    log.info(f"Advanced ratio= {advanced_ratio}")
+    log.info(f"Free stream velocity= {free_stream_velocity}")
+    log.info(f"Prandtl correction= {prandtl}")
+    log.info(f"Number of blades= {blades_number}")
+
+    # Resize the vectors using the number of radial stations.
+    r = np.empty(stations)
+    non_dimensional_radius = np.empty(stations)
+    dCp = np.empty(stations)
+    induced_velocity_distribution = np.empty(stations)
+    new_axial_interference_factor = np.empty(stations)
+    old_axial_interference_factor = np.empty(stations)
+    initial_axial_interference_factor = np.empty(stations)
+    dCt_new = np.empty(stations)
+    dCt_old = np.empty(stations)
+    dCt_0 = np.empty(stations)
+    delta_pressure = np.empty(stations)
+    optimal_axial_interference_factor = np.empty(stations)
+    optimal_rotational_interference_factor = np.empty(stations)
+    dCt_optimal = np.empty(stations)
+
+    non_dimensional_hub_radius = hub_radius / radius
+
+    # Computation of the non-dimensional radial stations.
+    for i in range(1, stations + 1):
+        r[i - 1] = i / float(stations)
+        if r[i - 1] <= non_dimensional_hub_radius:
+            i_hub = i - 1
```

Comment:
To be tested but I think it could work without loop, and it save you some variable declaration also:

```suggestion
    r = np.linspace(0, 1, stations)
    i_hub = np.abs(r - hub_radius / radius).argmin()
    if r[i_hub] < hub_radius:
       i_hub += 1
```